### DISCUSSION TITLE
Tasmota Decies Reboot -> IP SCAN

### DISCUSSION THREAD

**Discussion by ChrSch84** [arendst/Tasmota#21798] [Q&A]
Hello 

i have a strange Problem. 
A few of my Tasmota Devices reboot after making a IP Scan in my network. First i thought my device reboot (behind Tasmota) but after a while i found out if i make a ip scan with "http" in the Programm Advanced IP Scanner. The Tasmota devices reboots.

And the Strange thing is that is not the same Hardware! Mix of "Sonoff Pow", "NOUS A1T","EIGHTREE".

Here are the Settings from Advanced IP Scanner.
"
![image](https://github.com/user-attachments/assets/ed41bc2e-f590-4de6-81fb-ca0b8af1ba85)
"

Here a short List of my Devices: not all Reboot!

"
![image](https://github.com/user-attachments/assets/68e2c99c-c133-4627-87a2-2386a72251f8)
"

I could not find any hint why they reboot. 

Ideas? Hints?

Thanks for the Help.

------------------------------------------------------------------------------------------------------------

### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
Primitive tools like IP scanners may hammer the device with calls, and thereby overloading it. Some devices may not be able to power sustained Wifi activity when "spamming" by bad requests.

**Comment by ChrSch84** [comment]
But why not all Devices? And why did it work a few months back. Used this Scanner for Years and had no Problem with Tasmota.

And it is a bad thing if Tasmota could not handl wifi Spam and reboot the device!!!

ok if it reboot but only the software stack!

**Comment by sfromis** [comment]
Some devices have a better internal power supply than others, and can survive your DOS attack. And this is not about what Tasmota can handle, but about what your hardware can handle. Comparing back in time can be random factors, or just that devices degrade with time.

**Comment by ChrSch84** [comment]
- Old and new Devices are affekted.
- ITs an IP Scan not a DDoS Attack on the device.
- Its only if i select http/https all other "DDos" Attacks are fine.

**Comment by sfromis** [comment]
An IP scan is effectively a DOS attack. I did not say DDOS.

**Comment by ChrSch84** [comment]
ok fine DOS Attack...

Whatever...

but why is the device rebooting. If i make a "ARP" Request from the Router or Switch there is no Reboot. The IP Scan should be the same thing. And as i say its not every Tasmota Device!


So is it possible to Log the event? Or does someone know something about it?

**Comment by sfromis** [comment]
Well, that depends on what your crude tool is doing when it scans via http. It may be making a lot of requests. Way different from just an ARP request.

And when it is not every Tasmota device, that just supports my hypothesis that it is not about what Tasmota can handle, but about what weak hardware can handle. Thus, is not an "event" that Tasmota knows about, and even if it could log it, that log would be gone immediately when the device croaks.

**Comment by Jason2866** [comment]
How is the management tool you show getting the infos from the device?
mqtt or http? If it is http it is already stressing the devices. An additional load with a DOS tool can be to much for some devices.

**Comment by ChrSch84** [comment]
Crude Tool are you kidding me?

The only "Crude" thing here is you.

I ask if there is a Bug in the Firmware or a Setting. Mybe it is not only the "crude" IP Scan Tool (only HTTP/HTTPS).

Maybe it is a Bug in the Webserver ... Or something else?!

--------------------------

And what do you mean with WEAK Hardware? The Problem happens on an OLD Device and a new Device... Different Brand and Chip inside esp32 vs esp8266.

---------------------------

So in Short not very User Fiendly...


--------------------------------------------------------

The Management tool is only used AFTER the Problem. So no load on the http/s front.
All Communication is over mqtt.

**Reply by sfromis** [reply]
What I mean is that IP scanning via http is inherently a crude and not clearly defined thing to do. Not me or you, but such tools. And it is well known that weak hardware has trouble when pushing it with a lot of additional wifi activity. I consider it pretty friendly to tell you the facts, even if you may be starting with a different set of assumptions.

Still, *you* were the one talking about the device failing when you used the IP scanning tool. It is news that you now suggest a different story, not like what you posted to start the thread:

> A few of my Tasmota Devices reboot after making a IP Scan in my network. First i thought my device reboot (behind Tasmota) but after a while i found out if i make a ip scan with "http" in the Programm Advanced IP Scanner. The Tasmota devices reboots.

BTW. When I some time ago tried my own IP scanning via http, I made only a single http request per device, and saw no problems, even with Sonoff Basic devices which are on the weak end already.

**Comment by Jason2866** [comment]
Is this your usual friendly wording when people trying to help? Btw. probably by the most experienced Tasmota user. Well wish you luck with this way.

**Comment by ChrSch84** [comment]
@Jason2866

?? I dont start with the "friendly" wording.

Yeah i am not in a good mood if like @sfromis telling us users.: "dont use crude tool or weak hardware".
And then "closing" the Problem with the "hint" "Labels "[wont/can't fix]"

That is not helping.

Not me not you not Tasmota.

**Comment by ChrSch84** [comment]
So Fresh start:

Can someone recreate this problem?

**Reply by Jason2866** [reply]
Yes, as already mentioned with devices which have a borderline power supply. Since this is hardware related we can't change this in Tasmota firmware. Only way is "not to stress" this devices with heavy http queries which tools like IP scanners generates. The wording "crude" for many of this tools is correctly choosen since the act in the network like a hacking tool. Devices with security features enabled (like NAS and others) will add the source of the origin of suchi traffic on the ban list.
BTW. the label was added from me, and I moved to discussions (not closed) since no info added which a valid issue must have. All these are moved to discussions.
Discussions questions are answered as fast as issues.

**Reply by ChrSch84** [reply]
Ok i understand but what makes this a difference to any other Protokoll? That the Ip Scanner scans? It only happens @http or https level.

All other Information is collected?

And a bad Power Supply or a borderline power supply on a Socket Plug? Are they so close to the sun?

**Reply by Jason2866** [reply]
He, this is cheap China stuff. You get what you pay for. http is a heavy task for this little CPU. This means the CPU draws a lot more power than usual. The borderline power supply do fail here.
mqtt is a very lightweight protocol, made for such little devices. That's one of the reasons why Tasmota is using mqtt as communication protocol

The experience you have is an old story. It begun with the Sonoff Basic R2, which was the first device with such a "superior" power supply
All of the WiFi smart bulbs will have a short life. There is just not enough room for a power supply which will last over many years. The price is not possible when the needed small high quality components are used. Just that simple.

**Comment by arendst** [comment]
You might want to look at the result of command ``status 0`` on the restarted devices. It may show an exception. Look at several restarted devices and compare the exceptions and stack traces to find a common indication. Expect different traces though between different binaries and versions.
